File: proyectoestacionamiento/lib/Models/usuario.dart
class Usuario {
  final String correo;
  final String nombre;
  final String apellido;
  final String telefono;
  final String dni;
  final String fechaNacimiento;
  final String placa;
  final String modelo;

  Usuario({
    required this.correo,
    required this.nombre,
    required this.apellido,
    required this.telefono,
    required this.dni,
    required this.fechaNacimiento,
    required this.placa,
    required this.modelo,
  });

  /// Convertir JSON a objeto Usuario
  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      correo: json["correo"],
      nombre: json["nombre"],
      apellido: json["apellido"],
      telefono: json["telefono"],
      dni: json["dni"],
      fechaNacimiento: json["fecha_nacimiento"],
      placa: json["placa"],
      modelo: json["modelo"],
    );
  }

  /// Convertir objeto Usuario a JSON
  Map<String, dynamic> toJson() {
    return {
      "correo": correo,
      "nombre": nombre,
      "apellido": apellido,
      "telefono": telefono,
      "dni": dni,
      "fecha_nacimiento": fechaNacimiento,
      "placa": placa,
      "modelo": modelo,
    };
  }
}

File: backend_estacionamiento/backend/models/usuario.py
from pydantic import BaseModel, EmailStr

class UsuarioRegistro(BaseModel):
    correo: str

class UsuarioVerificacion(BaseModel):
    correo: str
    codigo: str

class UsuarioFormulario(BaseModel):
    correo: str
    nombre: str
    apellido: str
    telefono: str
    dni: str
    fecha_nacimiento: str  # formato "YYYY-MM-DD"
    placa: str
    modelo: str
    contrasenia: str

class UsuarioLogin(BaseModel):
    correo: str
    contrasenia: str

    # Modelo para el registro inicial de empresa
class EmpresaRegistro(BaseModel):
    nombre_empresa: str
    nombre_representante: str
    apellido_representante: str
    correo: EmailStr
    contrasenia: str
